controller/api/destination/endpoint_translator.go

```diff
@@ -205,30 +205,56 @@ func (et *endpointTranslator) NoEndpoints(exists bool) {
 	}
 }
 
+func (et *endpointTranslator) Update(set watcher.AddressSet) {
```

Comment:
semantically, what's the difference between Add and Update?  I think we'd be better off just using Add.  We'd need to update the logic in diffEndpoints to include endpoints in the Add set if they have changed from the previous update. (which is a good change anyway, imo)

Reply:
I added an additional check to the diffing logic in `diffEndpoints` that checks the old and new addresses. If they are different—as in the `OpaqueProtocol` field has changed—we now consider the address added.